src/fsharp/CheckFormatStrings.fs

```diff
@@ -366,7 +366,7 @@ let parseFormatStringInternal (m: range) (fragRanges: range list) (g: TcGlobals)
                   appendToDotnetFormatString "%"
                   parseLoop acc (i+1, fragLine, fragCol+1) fragments
 
-              | ('d' | 'i' | 'o' | 'u' | 'x' | 'X') ->
+              | ('d' | 'i' | 'u' | 'B' | 'o' | 'x' | 'X') when ch <> 'B' || cenv.g.langVersion.SupportsFeature LanguageFeature.PrintfBinaryFormat ->
```

Comment:
I think this would be better of the `'B'` case were split into its own fall-through/OR case on a new line with the guard. Otherwise it gets a little too difficult to read.

Reply:
Also, there is no `cenv` in scope here, as that's only within the compiler. Having this backed by the language flag might be more challenging because of that.

@KevinRansom @dsyme any thoughts?

Reply:
TcGlobals is there ... so g.langVersion.SupportsFeature LanguageFeature.PrintfBinaryFormat should work.

Reply:
Hmm, is it? I don't see any reference to it in here. Isn't the type defined in the compiler? I thought that the core library stuff was annotated with an attribute that we'd then wipe out when releasing a new version.

Reply:
Derp, it's there